pkg/webhooks/webhookpatch.go

```
@@ -97,53 +93,49 @@ func PatchCertLoop(injectionWebhookConfigName, webhookName, caBundlePath string,
+				go doPatchWithRetries(client, revision, config.Name, webhookName, caCertPem)
+			},
 		},
 	)
 	go controller.Run(stopCh)
+}
 
-	go func() {
-		var delayedRetryC <-chan time.Time
-		if retry {
-			delayedRetryC = time.After(delayedRetryTime)
-		}
-
-		for {
-			select {
-			case <-delayedRetryC:
-				if retry := doPatch(client, injectionWebhookConfigName, webhookName, caCertPem); retry {
-					delayedRetryC = time.After(delayedRetryTime)
-				} else {
-					log.Infof("Retried patch succeeded")
-					delayedRetryC = nil
-				}
-			case <-shouldPatch:
-				if retry := doPatch(client, injectionWebhookConfigName, webhookName, caCertPem); retry {
-					if delayedRetryC == nil {
-						delayedRetryC = time.After(delayedRetryTime)
-					}
-				} else {
-					delayedRetryC = nil
-				}
+func doPatchWithRetries(client kubernetes.Interface, revision, webhookConfigName, webhookName string, caCertPem []byte) {
+	retryDelay := time.Second * 2
+	for {
+		if err := doPatch(client, revision, webhookConfigName, webhookName, caCertPem); err != nil {
+			// if the webhook no longer has the target revision, bail out
+			if errors.Is(err, errWrongRevision) {
+				return
 			}
+
+			// otherwise, wait and give it another try
+			time.Sleep(retryDelay)
+		} else {
+			return
 		}
-	}()
+	}
 }
 
-func doPatch(cs kubernetes.Interface, webhookConfigName, webhookName string, caCertPem []byte) (retry bool) {
+func doPatch(cs kubernetes.Interface, revision, webhookConfigName, webhookName string, caCertPem []byte) error {
 	client := cs.AdmissionregistrationV1beta1().MutatingWebhookConfigurations()
-	if err := patchMutatingWebhookConfig(client, webhookConfigName, webhookName, caCertPem); err != nil {
-		log.Errorf("Patch webhook failed: %v", err)
-		return true
+	err := patchMutatingWebhookConfig(client, revision, webhookConfigName, webhookName, caCertPem)
+	if err != nil {
+		log.Errorf("Patch webhook failed for webhook %s: %v", webhookName, err)
```

this is super nitpicky, but I think standard go practice would be to return the err here. This ensures if we later add more code we are not accidentally ignoring the err

---

Removed the offending code but will keep this in mind for the future, thanks!